### PR TITLE
fix:(alert): updates alert

### LINKED ISSUE #1099: Require variant of Alert that includes Close button
We are missing an Alert variant that has a Close button. The ability to close an alert is a requirement for PF3 parity when the alert is used as an inline notification or a toast notification. For design guidance see the Alerts and Notifications design doc here: https://docs.google.com/document/d/1UH6J2Mf1Nqx4zUx-TzO6MCgPFq6NJatRb0kagmGN5yE/edit?usp=sharing

You can also see this capability for PF 3: 
https://www.patternfly.org/pattern-library/communication/inline-notifications/#code
https://www.patternfly.org/pattern-library/communication/toast-notifications/#code

### PR DESCRIPTION
**Pull request by matthewcarleton**
This closes #1099

### CODE REVIEW

File: src/patternfly/components/Alert/alert.scss [hunk 1]
```diff
@@ -87,25 +86,40 @@
...
   padding: var(--pf-c-alert__body--PaddingTop) var(--pf-c-alert__body--PaddingRight) var(--pf-c-alert__body--PaddingBottom) var(--pf-c-alert__body--PaddingLeft);
 }
 
 .pf-c-alert__title {
-  margin-top: var(--pf-c-alert__title--MarginTop);
-  margin-bottom: var(--pf-c-alert__title--MarginBottom);
   font-size: var(--pf-c-alert__title--FontSize);
   color: var(--pf-c-alert__title--Color);
+
+  & ~ * {
```
Comment (mcoker) [inline src/patternfly/components/Alert/alert.scss]: the `&` is redundant here

**Comment by mcoker** [comment]
Would we ever have both a close icon and an action button in an alert?

File: src/patternfly/components/Alert/alert.scss [hunk 2]
```diff
@@ -19,15 +18,15 @@
 
   // Title
   --pf-c-alert__title--FontSize: var(--pf-global--FontSize--md);
-  --pf-c-alert__title--MarginTop: var(--pf-global--spacer--xs);
-  --pf-c-alert__title--MarginBottom: var(--pf-global--spacer--xs);
   --pf-c-alert__title--Color: var(--pf-global--Color--100);
+  --pf-c-alert__title__siblings--MarginTop: var(--pf-global--spacer--xs);
```
[the remark this answers is not in the document]
Comment (matthewcarleton) [reply src/patternfly/components/Alert/alert.scss]: ya I was the same. I'm cool with `--`

File: src/patternfly/components/Alert/alert-dismiss.hbs [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="pf-c-alert__dismiss{{#if alert-dismiss--modifier}} {{alert-dismiss--modifier}}{{/if}}"
```
Comment (mcoker) [inline src/patternfly/components/Alert/alert-dismiss.hbs]: I think you can remove this file now

File: src/patternfly/components/Alert/examples/alert-variations-example.hbs [hunk 4]
```diff
@@ -2,6 +2,9 @@
   {{#> alert-icon alert-icon--success="true"}}
   {{/alert-icon}}
   {{#> alert-body alert-body--attribute='role="alert"'}}
+    {{#> button button--modifier="pf-m-plain" button--attribute='aria-label="Remove"'}}
```
Comment (mcoker) [inline src/patternfly/components/Alert/examples/alert-variations-example.hbs]: should the `aria-label` read "Close"?

Comment (matthewcarleton) [reply src/patternfly/components/Alert/examples/alert-variations-example.hbs]: oh hmm...maybe? Or maybe dismiss?

File: src/patternfly/components/Alert/examples/alert-types-example.hbs [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  {{/alert-icon}}
+  {{#> alert-body alert-body--attribute='role="alert"'}}
+    {{#> alert-title}}
+      {{#> screen-reader}}Info:{{/screen-reader}}
+      Info notification title
+    {{/alert-title}}
+  {{/alert-body}}
+{{/alert}}
+<br>
+{{#> alert alert--modifier="pf-m-success" alert--attribute='aria-label="success Notification"'}}
```
Comment (mcoker) [inline src/patternfly/components/Alert/examples/alert-types-example.hbs]: nitpick, can you capitalize "success Notification"?

**Pull request merged**